config/moved_config.go

```diff
@@ -14,13 +14,95 @@
 
 package config // import "go.opentelemetry.io/collector/config"
 
+import (
+	"go.opentelemetry.io/collector/component"
+)
+
 // Pipeline defines a single pipeline.
 // Deprecated: [v0.52.0] Use service.ConfigServicePipeline
 type Pipeline struct {
-	Receivers  []ComponentID `mapstructure:"receivers"`
-	Processors []ComponentID `mapstructure:"processors"`
-	Exporters  []ComponentID `mapstructure:"exporters"`
+	Receivers  []component.ID `mapstructure:"receivers"`
+	Processors []component.ID `mapstructure:"processors"`
+	Exporters  []component.ID `mapstructure:"exporters"`
 }
 
 // Deprecated: [v0.52.0] will be removed soon.
-type Pipelines = map[ComponentID]*Pipeline
+type Pipelines = map[component.ID]*Pipeline
+
+// Deprecated: [v0.64.0] use component.ReceiverConfig.
+type Receiver = component.ReceiverConfig
+
+// Deprecated: [v0.64.0] use component.UnmarshalReceiverConfig.
+var UnmarshalReceiver = component.UnmarshalReceiverConfig
+
+// Deprecated: [v0.64.0] use component.ReceiverConfigSettings.
+type ReceiverSettings = component.ReceiverConfigSettings
+
+// Deprecated: [v0.64.0] use component.NewReceiverConfigSettings.
+var NewReceiverSettings = component.NewReceiverConfigSettings
+
+// Deprecated: [v0.64.0] use component.ProcessorConfig.
+type Processor = component.ProcessorConfig
+
+// Deprecated: [v0.64.0] use component.UnmarshalProcessorConfig.
+var UnmarshalProcessor = component.UnmarshalProcessorConfig
+
+// Deprecated: [v0.64.0] use component.ProcessorConfigSettings.
+type ProcessorSettings = component.ProcessorConfigSettings
+
+// Deprecated: [v0.64.0] use component.NewProcessorConfigSettings.
+var NewProcessorSettings = component.NewProcessorConfigSettings
+
+// Deprecated: [v0.64.0] use component.ExporterConfig.
+type Exporter = component.ExporterConfig
+
+// Deprecated: [v0.64.0] use component.UnmarshalExporterConfig.
+var UnmarshalExporter = component.UnmarshalExporterConfig
+
+// Deprecated: [v0.64.0] use component.ExporterConfigSettings.
+type ExporterSettings = component.ExporterConfigSettings
+
+// Deprecated: [v0.64.0] use component.NewExporterConfigSettings.
+var NewExporterSettings = component.NewExporterConfigSettings
```

Comment:
Do you think we can simplify the new names in any way? Maybe reconsidering where we use `settings` vs `config`, or maybe even drop one of them? I'm not sure if we have strict definition for the terms to bring them from `config` package as is.

Reply:
Clearly a bad choice in terms of settings vs config. See also https://github.com/open-telemetry/opentelemetry-collector/pull/6422#discussion_r1009680107.

Short answer: for the moment we need to keep the long name to show that these settings "belong" to this config. We can debate in the future to remove this pattern completely.

Reply:
Ok, let's keep it consistent with existing pattern and reconsider later 👍

Reply:
Done.